app/controllers/application_controller.rb

```diff
@@ -5,8 +5,21 @@ class ApplicationController < ActionController::Base
   before_action :configure_permitted_parameters, if: :devise_controller?
 
   rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
-
+  include Pagy::Backend
   include Pundit
+  # with the use of authorise method, instance of the selected class becomes
+  # connected to the appropriate policy. In policy we have rules for each
+  # controller action, which determine scope and actions allowed to the user.
+
+  def index
+    @sample = Pokemon.all.sample(5)
+    @best_article = most_viewed
+  end
+
+  def show
+    @sample = Pokemon.all.sample(5)
+    @best_article = most_viewed
```

Comment:
Let's have one name for it, either `best_article` or `most_viewed`.
You can left a comment with description.